### ISSUE TITLE
Why is the attention_mask only used in self-attention, not used in cross-attention?

### ISSUE THREAD

**Issue by rockywind** [IDEA-opensource/DN-DETR#35]
Hi,
 Great work! I'm confused about why the attention mask is only used for self-attention? 如果去掉self-attention模块，只保留cross-attention模块，会不会造成noised boxes之间的信息泄露呢？
[https://github.com/IDEA-opensource/DN-DETR/blob/a59a5de5bf784f196e15bffed3145d05d5a9126a/models/DN_DAB_DETR/transformer.py#L125](url)
![image](https://user-images.githubusercontent.com/14838781/180836662-ce7740d8-ac10-4461-849b-51051750ce0b.png)
![image](https://user-images.githubusercontent.com/14838781/180836715-5364226f-6b78-4665-aa06-8069a90a4dfc.png)

**Comment by FengLi-ust** [comment]
We did not try this. Empirically cross-attention will not lead to information leakage.

**Comment by rockywind** [comment]
OK,thanks a lot!
